contracts/contracts/core/vault/src/main.sw

```diff
@@ -1549,123 +903,46 @@ fn _get_delta(
         _get_max_price(index_asset)
     };
 
-    let price_delta = if average_price > price {
-        average_price - price
-    } else {
-        price - average_price
-    };
-
-    let mut delta = size * price_delta / average_price;
-
-    let mut has_profit = false;
-    if is_long {
-        has_profit = price > average_price;
-    } else {
-        has_profit = average_price > price;
-    }
-
-    // if the minProfitTime has passed then there will be no min profit threshold
-    // the min profit threshold helps to prevent front-running issues
-    let min_bps = if get_unix_timestamp() > last_increased_time + storage.min_profit_time.read() {
-        0
+    let (has_profit, price_delta) = if price > average_price {
+        (is_long, price - average_price)
     } else {
-        storage.min_profit_basis_points.get(index_asset).try_read().unwrap_or(0)
+        (!is_long, average_price - price)
     };
 
-    if has_profit
-        && (delta * BASIS_POINTS_DIVISOR.as_u256()) <= (size * min_bps.as_u256())
-    {
-        delta = 0;
-    }
+    // TODO overflow check
+    let delta = size * price_delta / average_price;
 
     (has_profit, delta)
 }
 
 #[storage(read)]
 fn _get_position_fee(
     _account: Identity,
-    _collateral_asset: AssetId,
-    _index_asset: AssetId,
+    _index_asset: b256,
     _is_long: bool,
     size_delta: u256,
 ) -> u256 {
 
+    // TODO make fee u64
     if size_delta == 0 {
         return 0;
     }
 
-    let mut after_fee_usd = size_delta * (BASIS_POINTS_DIVISOR - storage.margin_fee_basis_points.read()).as_u256();
+    let mut after_fee_usd = size_delta * (BASIS_POINTS_DIVISOR - storage::vault.margin_fee_basis_points.read()).as_u256();
```

Comment:
Normally we would just do
`size_delta * margin_fee_basis_points / BASIS_POINTS_DIVISOR)`.

The present calculation is done in a way to round fees up. Please leave a comment on that.

Reply:
Agree. I changed this to your formula.